pkg/controller/sidecarset/sidecarset_processor.go

```diff
@@ -189,11 +200,46 @@ func (p *Processor) updatePodSidecarAndHash(control sidecarcontrol.SidecarContro
 	return err
 }
 
-func (p *Processor) listMatchedSidecarSets(pod *corev1.Pod) string {
+func (p *Processor) updatePodAnnotationsAboutSidecarNameAndRevision(control sidecarcontrol.SidecarControl, pod *corev1.Pod, latestRevision *apps.ControllerRevision) {
+	sidecarSet := control.GetSidecarset()
+	sidecarSetRevisions := make(map[string]string)
+	if revisions := pod.Annotations[sidecarcontrol.SidecarSetRevisionAnnotation]; len(revisions) > 0 {
+		if err := json.Unmarshal([]byte(revisions), &sidecarSetRevisions); err != nil {
+			klog.Errorf("parse pod(%s.%s) annotations[%s] value(%s) failed: %s",
+				pod.Namespace, pod.Name, sidecarcontrol.SidecarSetRevisionAnnotation, revisions, err.Error())
+		}
+	}
+
+	// older pods don't have SidecarSetListAnnotation
+	// older sidecarSets don't have controllerRevision
+	sidecarSetNames, nameOk := pod.Annotations[sidecarcontrol.SidecarSetListAnnotation]
+	sidecarCount := len(strings.Split(sidecarSetNames, ","))
+	if !nameOk || sidecarCount == 0 || len(sidecarSetRevisions) != sidecarCount {
+		sidecarSets := p.listMatchedSidecarSets(pod)
+		for _, sidecar := range sidecarSets {
+			// in the future, we will support revision control for sidecar injection,
+			// some pods may be injected old sidecarSet
+			if _, ok := sidecarSetRevisions[sidecar]; !ok {
+				sidecarSetRevisions[sidecar] = sidecarcontrol.SidecarLatestRevision
```

Comment:
Don't really understand what does the `SidecarLatestRevision` for?

Reply:
1. Some older pods may not have corresponding sidecar history revision, and they must the latest, so we use this to mark it.
2. If users want to check whether the latest sidecar has been injected, they do not need to check the historical revision number.

Reply:
I realize I'm wrong and I'll fix it as soon as possible.